lib/services/coursesV2.js

```diff
@@ -229,4 +229,58 @@ module.exports = class CoursesServiceV2 extends Schmervice.Service {
       return [errorHandler(err), null];
     }
   }
+
+  async createCourses(v1Courses) {
+    const { CoursesV2, Exercises } = this.server.models();
+    const { exercisesServiceV2 } = this.server.services();
+    // eslint-disable-next-line
+    for (const course of v1Courses) {
+      const courseSingle = _.pick(course, ['id', 'name', 'logo', 'short_description']);
+      courseSingle.lang_available = 'en';
+      if (courseSingle.lang_available) {
+        courseSingle.lang_available = val(courseSingle.lang_available).asArray().castTo('text[]');
+      }
+      // eslint-disable-next-line
+      await CoursesV2.query().insert(courseSingle);
+
+      const courseFolderName = `${courseSingle.name}_${courseSingle.id}`;
+      if (!fs.existsSync(`curriculum_new/${courseFolderName}`)) {
+        fs.mkdirSync(`curriculum_new/${courseFolderName}`);
+      }
+      // eslint-disable-next-line
+      const courseExercises = await Exercises.query()
+        .where('course_id', courseSingle.id)
+        .orderBy('sequence_num');
+      if (courseExercises !== []) {
+        // eslint-disable-next-line
+        for (const singleExercise of courseExercises) {
+          const exerciseData = {};
+          if (singleExercise.parent_exercise_id == null) {
+            exerciseData.name = singleExercise.name;
+          } else {
+            let newName = singleExercise.slug.split('__');
+            newName = newName[newName.length - 1].split('/').join('_').split('_');
+            newName = newName.slice(0, newName.length - 1).join('_');
+            exerciseData.name = newName;
+          }
```

Comment:
I think that all this splitting, joining, and slicing obscures how you're getting the `newName` and `filePath` from `singleExercise.slug`.  How about something like this instead?
```suggestion
          // slug_example: course_json__lessons__lesson1/part1_json
          exerciseData.name = singleExercise.parent_exercise_id == null ?
              singleExercise.name :
              // get between last '__' and '_' (lesson1/part1)
              singleExercise.slug.substring(singleExercise.slug.lastIndexOf('__') + 2,
              singleExercise.slug.lastIndexOf('_'))
              .replace('/\//g', '_'); // replace all / with _ (lesson1_part1)  
```
This assumes that there is at least one `__` and that if there is a `/` in `singleExercise.slug`, then the last `_` comes after the last `/`.